### PR TITLE
fix(request): correct delete permission check and await movie save

### PR DESCRIPTION
**Pull request by dougrathbone**
## Description

Two bugs in `server/routes/request.ts`:

**1. Anyone can delete any pending request**

The `DELETE /request/:id` handler used three top-level `&&` conditions to gate deletion:

```ts
// Before
if (
  !req.user?.hasPermission(Permission.MANAGE_REQUESTS) &&
  request.requestedBy.id !== req.user?.id &&
  request.status !== 1
) {
```

Because all three must be true to deny, the third condition short-circuits the check whenever a request is `PENDING` -- meaning any authenticated user can delete any pending request they did not create. The intended behaviour is that non-admins may only retract their own requests while they are still pending.

```ts
// After
if (
  !req.user?.hasPermission(Permission.MANAGE_REQUESTS) &&
  (request.requestedBy.id !== req.user?.id ||
    request.status !== MediaRequestStatus.PENDING)
) {
```

Also replaces the magic number `1` with `MediaRequestStatus.PENDING`.

**2. Movie request update is not awaited**

The movie branch of `PUT /request/:id` called `requestRepository.save()` without `await`, returning the `200` response before the write completed. The TV branch in the same handler correctly awaits the save.

**AI Disclosure:** I used Claude Code to help write the tests. I reviewed and understood all generated code before submitting.

## How Has This Been Tested?

New suite at `server/routes/request.test.ts` using the real SQLite test database via `setupTestDb()`:

- `allows the owner to delete their own pending request`
- `allows an admin to delete any pending request`
- `prevents a non-owner non-admin from deleting a pending request` -- directly tests the authorization bug
- `prevents the owner from deleting an approved request` -- asserts the status check is enforced
- `returns 404 for a non-existent request`
- `persists server and root folder changes to the database` -- regression guard for the unawaited movie save

**To reproduce bug 1 manually:**
1. Log in as a non-admin user (User A) and submit a request.
2. Log in as a different non-admin user (User B) who did not create the request.
3. Send `DELETE /api/v1/request/:id` as User B.
4. Before this fix: the request is deleted. After this fix: 403 Forbidden.

*This was independently confirmed by @fallenbagel via API before review.*

## Screenshots / Logs (if applicable)

N/A -- backend-only change with no UI impact.

## Checklist:

- [x] I have read and followed the contribution [guidelines](https://github.com/seerr-team/seerr/blob/develop/CONTRIBUTING.md).
- [x] Disclosed any use of AI (see our [policy](https://github.com/seerr-team/seerr/blob/develop/CONTRIBUTING.md#ai-assistance-notice))
- [x] I have updated the documentation accordingly. (no documentation changes required for this fix)
- [x] All new and existing tests passed.
- [x] Successful build `pnpm build`
- [x] Translation keys `pnpm i18n:extract` (no new UI strings added)
- [x] Database migration (if required) (no schema changes)

## Summary by CodeRabbit

* **Bug Fixes**
  * Tightened deletion authorization so requests can only be removed when allowed by status and ownership.
  * Ensured updates to movie requests are reliably persisted.

* **Tests**
  * Added end-to-end tests covering request deletion and update endpoints, including authorization and edge cases.

### CODE REVIEW

**Comment by dougrathbone** [comment]
@fallenbagel any chance i could get a review?

**Comment by fallenbagel** [comment]
Verified the bug via API as a non-admin user with a valid session cookie successfully deletes another user's pending request. Confirmed the permission check is broken as described.

Will review later.

**Review by fallenbagel — DISMISSED** [review]

**Comment by dougrathbone** [comment]
Thanks @fallenbagel ! Sorry for the nudge, but just happy to contribute to my favourite app for the first time. Keep on rocking! 🙇

File: server/routes/request.test.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    }
+  );
+  return app;
+}
+
+before(async () => {
+  app = createApp();
+  const settings = getSettings();
+  settings.main.localLogin = true;
+});
```
[the remark this answers is not in the document]
Comment (fallenbagel) [reply server/routes/request.test.ts:66]: Wouldn't the suggested fix still leak within the file as every `authenticatedAgent` call sets it back to true anyway, so the after hook just masks the issue rather than fixing it?

So I think it would be better and cleaner to scope the mutation to the helper itself by capturing and restoring the prior value around the login call:

```ts
async function authenticatedAgent(email: string, password: string) {
  const settings = getSettings();
  const priorLocalLogin = settings.main.localLogin;
  settings.main.localLogin = true;
  try {
    const agent = request.agent(app);
    const res = await agent.post('/auth/local').send({ email, password });
    assert.strictEqual(res.status, 200);
    return agent;
  } finally {
    settings.main.localLogin = priorLocalLogin;
  }
}
```

This way the setting is only true for the duration of the login call and restores correctly regardless of what it was before.  I have [another PR](https://github.com/seerr-team/seerr/pull/2824/changes#diff-1e6e65c0f01fe05f6b3b8a571d02e814fc6a22dbcf6008798555c0ad3173c5abR70-R83) that does exactly this, and a broader request route coverage PR in flight that uses this pattern as well.

[the remark this answers is not in the document]
Comment (fallenbagel) [reply server/routes/request.test.ts:66]: @dougrathbone I would recommend refactoring your unit test to follow the same pattern as: https://github.com/seerr-team/seerr/pull/2824/changes#diff-1e6e65c0f01fe05f6b3b8a571d02e814fc6a22dbcf6008798555c0ad3173c5ab 

So the above mentioned fix would apply as well. We should coordinate so we don't end up with conflicting changes down the line. I also have a full coverage of request route written (will create a draft PR later) which uses the same pattern

**Review by fallenbagel — CHANGES_REQUESTED** [review]

File: server/routes/request.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dougrathbone) [reply server/routes/request.test.ts:66]: Done - refactored the tests to align with #2824. Changes:

- Removed the global `settings.main.localLogin = true` from the `before` hook
- Renamed `authenticatedAgent` to `loginAs` with scoped capture-and-restore via `try/finally` (matching your pattern)
- Renamed `createMovieRequest` to `seedRequest` with the same signature as #2824
- Added the `sendNotification` mock and `beforeEach` reset
- Dropped `cookie: { secure: false }` from session config (fixes the CodeQL warning too)

Let me know if there's anything else to adjust - happy to coordinate on the request route coverage.

**Review by fallenbagel — APPROVED** [review]

**Review by gauthier-th — APPROVED** [review]
